openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py

```diff
@@ -0,0 +1,177 @@
+    """
+    Test to check that a user not registered in prerequisite is not counted while calculating minimal users
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(pre_req_courses)
+
+    assert minimal_users.count() == 0
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': True})], indirect=True)
+def test_minimal_users(enroll_user):
+    """
+    Test to check that an active user registered in prerequisite is counted while calculating minimal users
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(pre_req_courses)
+
+    assert minimal_users.count() > 0
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': True})], indirect=True)
+def test_user_failed_in_course(user_for_testing, enroll_user):
+    """
+    Test to check if the helper function returns True for a user not passed in course
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    is_user_failed = command.is_user_failed_in_course(user_for_testing, pre_req_courses.first())
+
+    assert is_user_failed == True
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': True})], indirect=True)
+def test_user_passed_in_course(mocker, user_for_testing, enroll_user):
+    """
+    Test to check if the helper function returns False for a user passed in course
+    """
+    mocker.patch.object(command_module.CourseGradeFactory, '_read')
+
+    class MockClass:
+        passed = True
+
+    mocker.return_value = MockClass()
+
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    is_user_failed = command.is_user_failed_in_course(user_for_testing, pre_req_courses.first())
+
+    assert is_user_failed == False
+
+
+@pytest.mark.django_db
+def test_handle_with_no_prerequisites():
+    """
+    Test to check if management command issues a system exits upon no prerequisites
+    """
+    with pytest.raises(SystemExit) as wrapped_exception:
+        call_command('update_is_prerequisite_courses_passed')
+
+    assert wrapped_exception.type == SystemExit
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': False})], indirect=True)
+def test_handle_with_no_minimal_user(user_for_testing, enroll_user):
+    """
+    Test to check if management command does not update `is_prerequisite_courses_passed` flag if minimal users are zero
+    """
+    call_command('update_is_prerequisite_courses_passed')
+
+    user_application_hub, _ = ApplicationHub.objects.get_or_create(user=user_for_testing)
+
+    assert user_application_hub.is_prerequisite_courses_passed == False
```

Comment:
Please mock `get_minimal_users_to_be_checked_for_update` and `is_user_failed_in_course` while writing test for core command, since these two methods are tested spertately